### ISSUE TITLE
I can't get xproto::set_input_focus() to work

### ISSUE THREAD

**Issue by sector-f** [rtbo/rust-xcb#13]
I'm currently trying to write a clone of wmutils' wtf command ([source](https://github.com/wmutils/core/blob/master/wtf.c)) in Rust. This program takes a window ID in hexadecimal as an argument and transfers focus to that window. My Rust version is available [here](https://github.com/sector-f/wmutils-rust/blob/master/src/wtf.rs). The relevant XCB call—the one that actually changes the focused window—is `xcb_set_input_focus()`.

I assume the Rust binding's equivalent to that function is `xproto::set_input_focus()`. They both take equivalent arguments; while the binding takes a `&Connection` rather than a `xcb_connection_t`, the functions seem to be essentially identical.

Not shown in my Rust version's code is my attempt to use the lower-level `ffi` module in place of the equivalent `xproto` function:
```rust 
unsafe {  ffi::xproto::xcb_set_input_focus(connection.get_raw_conn(), 1 as u8, win, 0);  }
```
with `connection` of course being of the `base::Connection` type. `ffi::xproto::XCB_INPUT_FOCUS_POINTER_ROOT` is a constant equal to 1, and `ffi::xproto::XCB_TIME_CURRENT_TIME` is a constant equal to 0, so using these values in place of the constants should not matter.
This attempt did not work either.

Additionally, I modified both the original C code and my Rust code to output the value of the window ID variable. They both return the same number, so I'm fairly certain that the problem is not an invalid window ID being passed to the function.

This leaves three possibilities that I can think of:

1. I'm doing something wrong
2. The binding isn't working
3. Both of the above

If there's any more information you need, just ask, and I'll answer to the best of my ability.

**Comment by rtbo** [comment]
Your code is correct.
this works:
```
unsafe {
    let conn = xcb_connect(null(), null_mut());
    xcb_set_input_focus(conn, XCB_INPUT_FOCUS_POINTER_ROOT as u8, win, XCB_CURRENT_TIME);
    xcb_flush(conn);
    xcb_disconnect(conn);
}
```
but this doesn't
```
let (conn, _) = xcb::Connection::connect(None).unwrap();
xcb::set_input_focus(&conn, xcb::INPUT_FOCUS_POINTER_ROOT as u8, win, xcb::CURRENT_TIME);
conn.flush();
```
(`connect` method in `0.7` takes an `Option<&str>` and return `Result`; also note that `xproto` is `pub use`d in `xcb`)

The only difference is that `xcb_prefetch_maximum_request_length` is called in the bindings. Inserting this call in the unsafe code and it doesn't work anymore.
I don't really understand why it blocks, but I don't mind removing this call from the bindings. This is more appropriate in application code anyway.

**Comment by panicbit** [comment]
@rtbo Oddly, adding a tiny delay at the end, even just `sleep_ms(1)` makes it work.

**Comment by sector-f** [comment]
@rtbo Well, here's the [official docs](https://xcb.freedesktop.org/manual/group__XCB__Core__API.html#ga365edf27111a8dd4e1d2831f652315e7) on `xcb_prefetch_maximum_request_length`.
What's jumping out at me is 

> Note that in order for this function to be fully non-blocking, the application must previously have called xcb_prefetch_extension_data(c, &xcb_big_requests_id) and the reply must have already arrived. 

Is that being called as well?

**Comment by rtbo** [comment]
@panicbit probably a race condition of some kind somewhere. 
@sector-f nope, it's not. I'll definitely remove this call.

I confirm that removing `xcb_prefetch_maximum_request_length` from the bindings makes your code work.

**Comment by sector-f** [comment]
Sounds good to me! Guess I'll just use `sleep()` for now though

**Comment by rtbo** [comment]
fixed with 76b8b39247019602738c40beda73eb5e8efdcc4d in v0.7.1
